pkg/ruler/ruler_ring.go

```diff
@@ -36,44 +33,22 @@ var RingOp = ring.NewOp([]ring.InstanceState{ring.ACTIVE}, func(s ring.InstanceS
 // is used to strip down the config to the minimum, and avoid confusion
 // to the user.
 type RingConfig struct {
-	KVStore          kv.Config     `yaml:"kvstore"`
-	HeartbeatPeriod  time.Duration `yaml:"heartbeat_period" category:"advanced"`
-	HeartbeatTimeout time.Duration `yaml:"heartbeat_timeout" category:"advanced"`
-
-	// Instance details
-	InstanceID             string   `yaml:"instance_id" doc:"default=<hostname>" category:"advanced"`
-	InstanceInterfaceNames []string `yaml:"instance_interface_names" doc:"default=[<private network interfaces>]"`
-	InstancePort           int      `yaml:"instance_port" category:"advanced"`
-	InstanceAddr           string   `yaml:"instance_addr" category:"advanced"`
-	NumTokens              int      `yaml:"num_tokens" category:"advanced"`
+	util.CommonRingConfig `yaml:",inline"`
```

Comment:
[nit] This is accessed via something like `cfg.CommonRingConfig.<xxx>`. We're already inside a config struct, so calling it `CommonRingConfig` is a bit weird. We could just use a name field called `CommonRing`.

If adopt this suggest, then do it everywhere for consistency.